### PR TITLE
Restore network configuration via kernel cli

### LINKED ISSUE #522: Fix network specification via QEMU command line
https://github.com/hermitcore/libhermit-rs/commit/eb2a46fe4d2892957a7e31fd8ce60bcd21a78f52 broke https://github.com/hermitcore/libhermit-rs/pull/337.

CC: @stlankes

### PR DESCRIPTION
**Pull request by JTS22**
Restores the ability to set the network IP address, gateway and subnet mask via the kernel commandline arguments `-ip`, `-gateway` and `-mask`.

Closes #522

### CODE REVIEW

**Review by mkroening — CHANGES_REQUESTED** [review]
Thanks for your PR! :)

It would be good to introduce an `fn var(key: &str) -> Option<String>` in `env` akin to [`std::env::var`](https://doc.rust-lang.org/nightly/std/env/fn.var.html).

The variables could then be defined with macros and with less redundancy, similar to how it was before https://github.com/hermitcore/libhermit-rs/commit/eb2a46fe4d2892957a7e31fd8ce60bcd21a78f52.

If you would rather have me do these changes, just say so.

Also, please squash your commits in the end. :)

**Comment by JTS22** [comment]
I agree that would make sense, I will try to implement that :)

**Review by mkroening — CHANGES_REQUESTED** [review]
Looks great! Just two minor suggestions.

File: src/syscalls/interfaces/mod.rs [hunk 1]
```diff
@@ -101,7 +101,7 @@ pub trait SyscallInterface: Send + Sync {
 		let envs = env::vars();
 		debug!("Setting envv as: {:?}", envs);
 		for a in envs {
-			let ptr = Box::leak(format!("{}\0", a).into_boxed_str()).as_ptr();
+			let ptr = Box::leak(format!("{}={}\0", a.0, a.1).into_boxed_str()).as_ptr();
```
[the remark this answers is not in the document]
Comment (JTS22) [reply src/syscalls/interfaces/mod.rs]: Done

File: src/env.rs [hunk 2]
```diff
@@ -95,8 +99,13 @@ pub fn freq() -> Option<u16> {
 	CLI.get().unwrap().freq
 }
 
-pub fn vars() -> &'static [String] {
-	CLI.get().unwrap().env_vars.as_slice()
+#[allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (JTS22) [reply src/env.rs]: Changed it, though clippy does not seem to like it...

[the remark this answers is not in the document]
Comment (mkroening) [reply src/env.rs]: Ah, yes. Looking at it more closely, it should be `cfg(all(feature = "tcp", not(feature = "dhcpv4")))`, I think.

[the remark this answers is not in the document]
Comment (JTS22) [reply src/env.rs]: Yes, that works :)

**Pull request merged**